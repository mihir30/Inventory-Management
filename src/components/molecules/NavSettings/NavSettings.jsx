import React from "react";
import { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import "./NavSettings.scss";
import { getData } from "../../utils/common.utils";

const NavSettings = ({ ...props }) => {
  let navigate = useNavigate();
  const plpURL = window.location.href;
  let extractCategoryID = plpURL.split("/");
  let extractCategoryIDLen = extractCategoryID.length;
  if (extractCategoryID[extractCategoryIDLen - 1] == '') {
    extractCategoryID = extractCategoryID[extractCategoryIDLen - 2];
  }
  else {
    extractCategoryID = extractCategoryID[extractCategoryIDLen - 1];
  }

  const [itemList, setItemList] = useState([]);
  useEffect(() => {
    const fetchData = async () => {
      try {
        const items = await getData("http://localhost:8000/equipments");
        setItemList(items);
      } catch (error) {
        console.log("error", error.message);
      }
    };
    fetchData();
  }, [plpURL]);

  const AddEquipment = () => {
    navigate("/unittypes/0");
  };
  const [inputText, setInputText] = useState("");
  const [filteredResults, setFilteredResults] = useState([]);

  const handleChange = (e) => {
    setInputText(e.target.value.toLowerCase());
  };

  useEffect(() => {
    if (inputText !== "") {
      const filteredData = itemList.filter((item) => {
        return Object.values(item.name)
          .join("")
          .toLowerCase()
          .startsWith(inputText.toLowerCase());
      });
      setFilteredResults(filteredData);
    } else {
      setFilteredResults(itemList);
    }
  }, [inputText, itemList]);

  return (
    <div className="category-container">
      <ul className="category-list">
        <li className="search-add-bar">
          <input
            className="search-box-settings"
            type="text"
            aria-label="search-equipment"
            onChange={handleChange}
          />

          <img
            src="https://cdn0.iconfinder.com/data/icons/multimedia-solid-30px/30/search-256.png"
            alt="new"
            className="search-icon"
          />
          <img
            src="https://cdn0.iconfinder.com/data/icons/interface-line-4/48/Add_circle_new-256.png"
            alt="new"
            className="add-equipment"
            onClick={AddEquipment}
          />
        </li>

        {itemList && itemList.length > 0
          ? inputText.length > 0
            ? filteredResults.map((item) => {
              return (
                <>
                  <li
                    key={item.id}
                    id={
                      item.archive == true ? "archive-nav" : ""
                    }
                    className={
                      item.id == extractCategoryID && extractCategoryID != 0
                        ? "equipments-active"
                        : "equipments"
                    }
                    onClick={() => props.handleCategoryChange(item.id)}
                  >
                    <img
                      src={item.productIconURL}
                      className="equip-icon"
                      alt="equipment-icons"
                    ></img>
                    {item.name}
                  </li>
                </>
              );
            })
            : itemList.map((item) => {
              return (
                <>
                  <li
                    key={item.id}
                    id={
                      item.archive == true ? "archive-nav" : ""
                    }
                    className={
                      item.id == extractCategoryID && extractCategoryID != 0
                        ? "equipments-active"
                        : "equipments"
                    }
                    onClick={() => props.handleCategoryChange(item.id)}
                  >
                    <img
                      src={item.productIconURL}
                      className="equip-icon"
                      alt="equipment-icons"
                    ></img>
                    {item.name}
                  </li>
                </>
              );
            })
          : null}
      </ul>
    </div>
  );
};

export default NavSettings;
