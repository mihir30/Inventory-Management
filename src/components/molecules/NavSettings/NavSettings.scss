.category-container {
    background-color: #EBEDF0;
    width: 210px;
    height: 1280px;

    .search-add-bar {
        padding: 3px 1px 10px 10px;
        border-bottom: 1px solid #edeff0;
        border-color: #DADEE0;
    }

    .search-box-settings {
        border-radius: 2px;
        width: 150px;
        height: 30px;
        outline: 0;
        border: 1px solid #ccd7e6;
        border-radius: 0;
        color: #475466;
        font-family: Roboto, sans-serif;
        font-size: 14px;
        padding: 0 9pt;
    }

    .search-box-1:focus {
        border: 1px solid #1ca1c1;
    }


    .search-icon {
        height: 20px;
        width: 20px;
        position: relative;
        right: 25px;
    }

    .add-equipment {

        width: 25px;
        height: 25px;
        position: relative;
        top: 3px;
        right: 10px;

    }

    .equip-icon {
        width: 25px;
        margin-right: 15px;
        height: 25px;
    }

    .category-list {
        list-style-type: none;
        font-size: 1rem;
        padding: 0;
        margin: 0;
        cursor: pointer;

        .equipments {
            border-bottom: 1px solid #edeff0;
            border-color: #DADEE0;
            padding: 1rem;
            color: #475466;
            font-size: 14px;
        }

        .equipments:hover {
            background: #f4f5f9;
            border-bottom: 1px solid #edeff0;
            border-color: #DADEE0;
            padding: 1rem;
            color: #475466;
            font-size: 14px;
            box-shadow: inset 3px 0 #1ca1c1;

        }

        .equipments-active {
            background: #f4f5f9;
            border-bottom: 1px solid #edeff0;
            border-color: #DADEE0;
            padding: 1rem;
            color: #475466;
            font-size: 14px;
            box-shadow: inset 3px 0 #1ca1c1;
        }

        #archive-nav {
            border-bottom: 1px solid #edeff0;
            border-color: #DADEE0;
            padding: 1rem;
            color: #94A1B3;
            font-size: 14px;
        }
    }
}