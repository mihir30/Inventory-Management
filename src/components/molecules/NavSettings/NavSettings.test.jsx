import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import NavSettings from './NavSettings';

describe("Nav settings Component", () => {
    it('renders the component properly', () => {
        const divEl = document.createElement("div");
        const handleCategoryChange = jest.fn();
        ReactDom.render(<BrowserRouter><NavSettings
            handleCategoryChange={handleCategoryChange}
          ></NavSettings></BrowserRouter>, divEl); 
        
    });


    it('should have the right nav settings component snapshot', () => {
        const tree = renderer.create().toJSON();
        expect(tree).toMatchSnapshot();
    })

});