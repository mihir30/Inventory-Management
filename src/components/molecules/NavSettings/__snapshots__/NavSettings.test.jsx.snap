// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav settings Component should have the right nav settings component snapshot 1`] = `null`;
